codegen/smithy-aws-swift-codegen/src/main/kotlin/software/amazon/smithy/aws/swift/codegen/awsquery/AwsQueryProtocolGenerator.kt

```diff
@@ -38,22 +41,33 @@ open class AwsQueryProtocolGenerator : AWSHttpBindingProtocolGenerator() {
     override val httpResponseGenerator = HttpResponseGenerator(
         unknownServiceErrorSymbol,
         defaultTimestampFormat,
-        AWSRestXMLHttpResponseBindingErrorGenerator(),
+        XMLHttpResponseBindingOutputGenerator(),
+        AWSQueryHttpResponseBindingErrorGenerator(),
```

Comment:
A little refactoring here, plus code is modified to call XML variants of various generators when appropriate.